### PR TITLE
Layout has repr

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
Makes Layout have a `repr`:

```python

Layout({
(QuantumRegister(14, 'q'), 0): 0,
(QuantumRegister(14, 'q'), 1): 1,
(QuantumRegister(14, 'q'), 2): 2,
(QuantumRegister(14, 'q'), 3): 13,
(QuantumRegister(14, 'q'), 4): 3,
(QuantumRegister(14, 'q'), 5): 12,
(QuantumRegister(14, 'q'), 6): 4,
(QuantumRegister(14, 'q'), 7): 11,
(QuantumRegister(14, 'q'), 8): 10,
(QuantumRegister(14, 'q'), 9): 5,
(QuantumRegister(14, 'q'), 10): 9,
(QuantumRegister(14, 'q'), 11): 6,
(QuantumRegister(14, 'q'), 12): 8,
(QuantumRegister(14, 'q'), 13): 7,
})
```

### CODE REVIEW

**Comment by nonhermitian** [comment]
Not much love for `repr`.

**Comment by 1ucian0** [comment]
I can take a look to this. But I have not ranking high in priority. If you dont mind, we can keep it open until I (or somebody else) get a free slot.

File: qiskit/mapper/layout.py [hunk 1]
```diff
@@ -30,6 +30,13 @@ def __init__(self, input_=None):
         if isinstance(input_, list):
             self.from_list(input_)
 
+    def __repr__(self):
+        """Representation of a Layout"""
+        str_list = []
+        for key, val in self._v2p.items():
+            str_list.append("{k}: {v},".format(k=key, v=val))
+        return "Layout({\n"+"\n".join(str_list)+"\n})"
```
Comment (1ucian0) [inline qiskit/mapper/layout.py]: Maybe avoid the last comma?
```python
            str_list.append("{k}: {v}".format(k=key, v=val))
        return "Layout({\n"+",\n".join(str_list)+"\n})"
```

**Comment by 1ucian0** [comment]
`test.python.quantum_info.channel.test_linearops.TestEquivalence.test_kraus_multiply` is failing randomly https://travis-ci.com/Qiskit/qiskit-terra/jobs/188081821 Shall I re-run or do you want to have a look?

**Comment by nonhermitian** [comment]
The errors are unrelated, and are causing other PR failures as well. Please just rerun. I cannot do it.

**Pull request merged**